lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -170,8 +170,11 @@ def add_generated_directed_dependency(parent, child, label=nil)
     }
 
     if not edge_exists
+      parameters = target.original_parameters
+      resource = ::Puppet::Resource.new(target.class, target.title, :parameters => parameters)
```

Comment:
I had split it for debug prints, and never put it back together. I don't have strong opinions about the format here, I'm hapy to change it.